src/Gambling.js

```diff
@@ -1,12 +1,233 @@
+  const [result, setResult] = useState(null);
+
+  let provider, signer, contract;
+
+  // Requests access to the user's Meta Mask Account
+  async function requestAccount() {
+    await window.ethereum.request({ method: "eth_requestAccounts" });
+  }
+
+  useEffect(() => {
+    handleShowInfo();
+  }, []);
+
+  // useEffect(() => {
+  //   const provider = new ethers.providers.Web3Provider(window.ethereum);
+  //   const contract = new ethers.Contract(donationContractAddress, DonationContract.abi, provider);
+
+  //   // Subscribe to the ValueChanged event
+  //   contract.on("FundsTransferredToUser", (from, newValue) => {
+  //     setEventData(prevEventData => [...prevEventData, { from, newValue }]);
+  //   });
+
+  //   // Cleanup function to unsubscribe from the event listener
+  //   return () => {
+  //     contract.removeAllListeners("FundsTransferredToUser");
+  //   };
+  // }, []);
+
+
+  // show money_pool
+  async function handleShowInfo() {
+    provider = new ethers.providers.Web3Provider(window.ethereum);
+    signer = provider.getSigner();
+    contract = new ethers.Contract(gamblingAddress, Gambling.abi, signer);
+
+    const amount = await contract.getMoneyPoolBalance();
+    const amount_f = ethers.utils.formatEther(amount);
+    const balance = await contract.getUserBalance();
+    const balance_f = ethers.utils.formatEther(balance);
+    setMoneyPool(amount_f);
+    setUserBalance(balance_f);
+  }
+
+  //deposit
+  async function handleDeposit() {
+    if (!depositAmount) return;
+    try {
+      if (selectedPre !== 0) {
+        if (typeof window.ethereum !== "undefined") {
+          await requestAccount();
+          const provider = new ethers.providers.Web3Provider(window.ethereum)
+          const signer = provider.getSigner();
+          const contract = new ethers.Contract(gamblingAddress, Gambling.abi, signer);
+
+          const ethValue = ethers.utils.parseEther(depositAmount);
+          const depositTx = await contract.depositAndPredict(selectedPre, { value: ethValue });
+          await depositTx.wait();
+          setDepositAmount("")
+          handleShowInfo();
+        }
+      } else {
+        alert("you need to select your prediction first");
+      }
+    } catch (error) {
+      console.error(error);
+    }
+  }
+
+
+  async function handlResult() {
+    try {
+      if (typeof window.ethereum !== "undefined") {
+        await requestAccount();
+        const provider = new ethers.providers.Web3Provider(window.ethereum)
+        const signer = provider.getSigner();
+        const contract = new ethers.Contract(gamblingAddress, Gambling.abi, signer);
+
+        const result = await contract.revealResult();
+        await result.wait();
+        setResult(result);
+
+        handleShowInfo();
+      }
+    } catch (error) {
+      console.error(error);
+    }
+  };
+
+  // Fetches the current value store in greeting
+  async function fetchGreeting() {
```

Comment:
Please update function name, be more meaningful like fetchBTCPrice or fetchETHPrice